frontend/test/pytest/test_mid_circuit_measurement.py

```diff
@@ -210,6 +212,256 @@ def circuit(x):
         assert circuit(0.0) == [0] * 10
         assert circuit(jnp.pi) == [1] * 10
 
+    def test_dynamic_one_shot_with_sample_single(self, backend):
+        """Test that a measurement result can be returned with qml.sample and shots."""
+
+        dev = qml.device(backend, wires=1, shots=10)
+
+        @qjit
+        @dynamic_one_shot
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RY(x, wires=0)
+            m = measure(0)
+            qml.PauliX(0)
+            return qml.sample(m)
+
+        assert jnp.allclose(circuit(0.0), 0)
+        assert jnp.allclose(circuit(jnp.pi), 1)
```

Comment:
Should these not be arrays of 10 values?

Reply:
`allclose` will broadcast. Are you concerned we are not validating the shapes?

Reply:
Oh nice, didn't know the broadcasting thing. Validating the shape in addition is a good idea I think